lib/Os.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * @author Frank Karlitschek <frank@nextcloud.com>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ServerInfo;
+
+use bantu\IniGetWrapper\IniGetWrapper;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IL10N;
+use OCA\ServerInfo\OperatingSystems\DefaultOs;
+
+class Os {
+
+	/** @var IClientService */
+	protected $clientService;
+
+	/** @var IConfig */
+	protected $config;
+
+	/** @var IDBConnection */
+	protected $connection;
+
+	/** @var IniGetWrapper */
+	protected $phpIni;
+
+	/** @var \OCP\IL10N */
+	protected $l;
+
+	/** @var */
+	protected $backend;
+
+	/** @var */
+	protected $osname;
+
+	/**
+	 * Os constructor.
+	 *
+	 * @param IClientService $clientService
+	 * @param IConfig $config
+	 * @param IDBConnection $connection
+	 * @param IniGetWrapper $phpIni
+	 * @param IL10N $l
+	 */
+	public function __construct(IClientService $clientService, IConfig $config, IDBConnection $connection, IniGetWrapper $phpIni, IL10N $l) {
+		$this->clientService = $clientService;
+		$this->config = $config;
+		$this->connection = $connection;
+		$this->phpIni = $phpIni;
+		$this->l = $l;
+
+		$detectedOs = $this -> detectOs();
+		switch ($detectedOs) {
+			case 'ubuntu':
+				$this->backend = new DefaultOs();
+				break;
+			default:
+				$this->backend = new DefaultOs();
+				break;
+		}
+	}
+
+
+	/**
+	 * @return string
+	 */
+	public function detectOs() {
```

Comment:
use this:

```suggestion
	public function detectOs(): string {
```

instead of the doc block, if the doc block doesn't have more details.